jenkins/automation/splint-memory-leak.groovy

```diff
@@ -0,0 +1,93 @@
+        string(name: 'CORTX_MOTR_REPO', defaultValue: 'https://github.com/Seagate/cortx-motr/', description: 'Repository for Memory Leak Analysis', trim: true)
+    }
+
+    stages {
+        stage('Checkout Script') {
+            steps {
+                cleanWs()
+                script {
+                    checkout([$class: 'GitSCM', branches: [[name: "${CORTX_MOTR_BRANCH}"]], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: "${CORTX_MOTR_REPO}"]]])
+                }
+            }
+        }
+
+        stage ('Execute Memory Analysis') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                sh label: '', returnStatus: true, script: '''
+                yum install splint -y
+                mkdir -p artifacts
+                for file in  $(find $PWD -type f -regextype posix-egrep -regex ".+\\\\.(c|h|lcl)$"); do splint +trytorecover -preproc -warnposix $file >> artifacts/splint-analysis.log ; done || echo "Log Generated"
+                '''
+            }
+        }
+    }
+
+    post {
+
+        cleanup {
+                script {
+                    // Archive Deployment artifacts in jenkins build
+                    archiveArtifacts artifacts: "artifacts/*.*", onlyIfSuccessful: false, allowEmptyArchive: true
+                }
+        }
+
+        always {
+            script {
+                // Jenkins Summary
+                clusterStatus = ""
+                if ( currentBuild.currentResult == "SUCCESS" ) {
+                    MESSAGE = "Memory Leak Analysis Execution Success for the build ${build_id}"
+                    ICON = "accept.gif"
+                    STATUS = "SUCCESS"
+                } else if ( currentBuild.currentResult == "FAILURE" ) {
+                    manager.buildFailure()
+                    MESSAGE = "Memory Leak Analysis Execution Failed for the build ${build_id}"
+                    ICON = "error.gif"
+                    STATUS = "FAILURE"
+
+                } else {
+                    manager.buildUnstable()
+                    MESSAGE = "Memory Leak Analysis Execution is Unstable for the build ${build_id}"
+                    ICON = "warning.gif"
+                    STATUS = "UNSTABLE"
+                }
+
+                MemoryLeakAnalysisStatusHTML = "<pre>${clusterStatus}</pre>"
+
+                manager.createSummary("${ICON}").appendText("<h3>Memory Leak Analysis Execution ${currentBuild.currentResult} </h3><p>Please check <a href=\"${BUILD_URL}/artifact/artifacts/splint-analysis.log\">Memory Leak Analysis logs</a> for Memory leak errors</p><p>Please check <a href=\"${BUILD_URL}/console\">Memory Leak Analysis Execution logs</a> for more info</p>", false, false, false, "red")
+
+                // Email Notification
+                env.build_stage = "${build_stage}"
+                env.cluster_status = "${MemoryLeakAnalysisStatusHTML}"
```

Comment:
can you change variable name to appropriate name on what it holds?

Reply:
We will store metrics collected by parsing raw splint tool logs in MemoryLeakAnalysisStatusHTML variable

Reply:
@shailesh-vaidya, can we change variable `env.cluster_status` as well to something like `env.memory_leak_stats`

Reply:
Fixed and tested in https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/Motr/job/memory-leak-analysis/19/